lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -317,11 +317,38 @@ class LLDBTypeInfoProvider : public swift::remote::TypeInfoProvider {
     std::string wrapped;
     // The mangled name passed in is bare. Add global prefix ($s) and type (D).
     llvm::raw_string_ostream(wrapped) << "$s" << mangledName << 'D';
+    swift::Demangle::Demangler dem;
+    auto *node = dem.demangleSymbol(wrapped);
+    if (!node) {
+      // Try `mangledName` as plain ObjC class name.
+      auto *global = dem.createNode(Node::Kind::Global);
```

Comment:
Can we move this logic into TypeSystemSwiftTypeRef?

Reply:
factored out a couple functions into `SwiftDemangle.h`